File: platform/models/sessions.ts
import { Event } from "./events";
import { Task } from "./tasks";

export interface Session {
  id: string;
  created_at: number;
  project_id: string;
  metadata?: Record<string, any>;
  data?: Record<string, any>;
  preview?: string;
  environment?: string;
  notes?: string;
}

export interface SessionWithEvents extends Session {
  events: Event[];
}
export interface SessionWithTasks extends Session {
  tasks: Task[];
}

File: backend/phospho_backend/api/v2/models/sessions.py
from pydantic import BaseModel

from phospho_backend.db.models import Session


class Sessions(BaseModel):
    sessions: list[Session]


class SessionCreationRequest(BaseModel):
    project_id: str
    data: dict | None = None


class SessionUpdateRequest(BaseModel):
    metadata: dict | None = None
    data: dict | None = None
    preview: str | None = None
    notes: str | None = None
